firefox-ios/Client/Frontend/Home/Homepage Rebuild/HomepageViewController.swift

```diff
@@ -690,25 +710,39 @@ final class HomepageViewController: UIViewController,
                     actionType: PocketActionType.enteredForeground
                 )
             )
-        case .ProfileDidFinishSyncing,
-                .PrivateDataClearedHistory,
-                .FirefoxAccountChanged,
+        case .PrivateDataClearedHistory,
                 .TopSitesUpdated,
                 .DefaultSearchEngineUpdated:
-            store.dispatch(
-                TopSitesAction(
-                    windowUUID: self.windowUUID,
-                    actionType: TopSitesActionType.fetchTopSites
-                )
-            )
+            dispatchActionToFetchTopSites()
         case .BookmarksUpdated, .RustPlacesOpened:
             store.dispatch(
                 BookmarksAction(
                     windowUUID: self.windowUUID,
                     actionType: BookmarksActionType.fetchBookmarks
                 )
             )
+        case .ProfileDidFinishSyncing, .FirefoxAccountChanged:
+            dispatchActionToFetchTopSites()
+            dispatchActionToFetchRemoteTabs()
         default: break
         }
     }
+
+    private func dispatchActionToFetchTopSites() {
+        store.dispatch(
+            TopSitesAction(
+                windowUUID: self.windowUUID,
+                actionType: TopSitesActionType.fetchTopSites
+            )
+        )
+    }
+
+    private func dispatchActionToFetchRemoteTabs() {
+        store.dispatch(
+            JumpBackInAction(
+                windowUUID: self.windowUUID,
+                actionType: JumpBackInActionType.fetchRemoteTabs
+            )
+        )
+    }
```

Comment:
I realize that these action types are based on the consequences of the action vs the action taken. As opposed to adding additional changes to this PR, I'll create another one to address the changes. 

Right now each section has a specific action tied to it, but to follow the Redux pattern, we should add them to the homepage action types. Can discuss more on that PR.